language/move-lang/src/parser/lexer.rs

```diff
@@ -205,6 +209,152 @@ impl<'input> Lexer<'input> {
         self.prev_end
     }
 
+    fn offset_while(
+        &mut self,
+        offset: usize,
+        f: impl Fn(Symbol, usize, char) -> Result<bool, Diagnostic>,
+    ) -> Result<usize, Diagnostic> {
+        let mut result = offset;
+        let mut chars = self.text[result..].char_indices();
+        while let Some((i, c)) = chars.next() {
+            let idx = offset + i;
+            match f(self.file, idx, c) {
```

Comment:
Who am I, why am I commenting this? I normally would always argue for pattern matching but...
This might be better as just `if f(self.file, idx, c)? { ... } else { ... }` 
Thoughts?